python_modules/dagster/dagster_tests/definitions_tests/decorators_tests/test_asset_decorator_with_check_specs.py

```diff
@@ -722,3 +722,137 @@ def foo(context: AssetExecutionContext):
 
     assert check_eval.asset_key == AssetKey("asset1")
     assert check_eval.check_name == "check1"
+
+
+@op
+def create_asset():
+    return None
+
+
+@op
+def validate_asset(word):
+    return AssetCheckResult(check_name="check1", passed=True, metadata={"foo": "bar"})
+
+
+@graph_multi_asset(
+    outs={"asset_one": AssetOut(), "asset_two": AssetOut()},
+    check_specs=[AssetCheckSpec("check1", asset="asset_one", description="desc")],
+    can_subset=True,
+)
+def my_asset():
+    asset_one = create_asset()
+    return {
+        "asset_one": asset_one,
+        "asset_one_check1": validate_asset(asset_one),
+        "asset_two": create_asset(),
+    }
+
+
+def test_graph_asset_all():
+    result = materialize(assets=[my_asset])
+    assert result.success
+
+    assert len(result.get_asset_materialization_events()) == 2
+    assert len(result.get_asset_check_evaluations()) == 1
+
+
+def test_graph_asset_subset_no_checks():
+    result = materialize(
```

Comment:
My understanding is that it's currently not possible to execute checks alone without executing the corresponding asset, is that correct?

If not, this would be a good case to test.

Reply:
https://github.com/dagster-io/dagster/pull/20115/files#diff-66c934e188c377f74363d3f1764d1a8de0a96001b69edd54e54821d53549ffe2R1542 prevents just checks from being launched from host processes. Ideally we'd unblock this but I think it can wait, I'll file a tracking issue